CoreUtilities/src/au/gov/asd/tac/constellation/utilities/genericjsonio/JsonIO.java

```diff
@@ -0,0 +1,289 @@
+     * @param filePrefix Ignored if blank, if not, a string to prefix the
+     * filename with.
+     *
+     */
+    public static void saveJsonPreferences(String saveDir, ObjectMapper mapper, ArrayNode rootNode, String filePrefix) {
+        // Normal operation sees an empty filePrefix string supplied. In this case, the name of the file will contain
+        // only the filename entered by the user. If however a non-empty filePrefix is supplied, then the filename
+        // privided by the user will be prefixed with this.
+        // This is MOST useful when a dedicated directory exists for similar 'type' config files, like the example
+        // directory structure below:
+        // <user-dir>
+        //     +-- tableconfigs
+        //              +--- transaction_config1.json
+        //              +--- transaction_config2.json
+        //              +--- transaction_config3.json
+        //              +--- vertex_configa.json
+        //              +--- vertex_configb.json
+        //              +--- vertex_configc.json
+        //
+        // The above structure could have been contructed with multiple calls to saveJsonPreferences all using
+        // tableconfigs as the value for saveDir, three using 'transaction_' for filePrefix and the other three
+        // using 'vertex_'.
+        // This functionality is integrated with loadJsonPreferences such that if a filePrefix is supplied in the
+        // loadJsonPreferences call then only files that contain the prefix are offered to the user to load. Hence,
+        // it allows a form of config file filtering.
+        final Preferences prefs = NbPreferences.forModule(ApplicationPreferenceKeys.class);
+        final File prefDir = new File(ApplicationPreferenceKeys.getUserDir(prefs), saveDir);
+
+        // Create containing directory if it doesn't exist, wnsure it was successful.
+        if (!prefDir.exists()) {
+            prefDir.mkdir();
+        }
+        if (!prefDir.isDirectory()) {
+            final String msg = String.format("Can't create data access directory '%s'.", prefDir);
+            final NotifyDescriptor nd = new NotifyDescriptor.Message(msg, NotifyDescriptor.ERROR_MESSAGE);
+            DialogDisplayer.getDefault().notify(nd);
+        }
+
+        // Configure the mapper
+        mapper.configure(SerializationFeature.INDENT_OUTPUT, true);
+        mapper.configure(SerializationFeature.CLOSE_CLOSEABLE, true);
+
+        // Obtain a filename from the user. If no filename is supplied, and the user didn't hit cancel, then
+        // generate a filename for them based on username and timestamp. If cancel was hit, no more processing
+        // is required.
+        Tuple<Boolean, String> preferenceNameDetails = JsonIODialog.getName();
+        if (!preferenceNameDetails.getFirst()) {
+            // Cancel was pressed, lets exit straight away - nothing to do here
+            return;
```

Comment:
is there a way to right the void function without the return? More that sonar is probably going to flag this

Reply:
I hadn't seen sonar complain about this, but can inverse the test and wrap all the remaining function code in the if condition if you feel its worth addressing.

Reply:
@antares1470 and @serpens24 based on a test today #401 I didn't see SonarQube complain about this.

Reply:
In that case I'll leve as is... I think it reads quite simply... sometimes easier than nested if conditions.